src/pydata_sphinx_theme/toctree.py

```diff
@@ -36,7 +37,25 @@ def add_toctree_functions(
     """Add functions so Jinja templates can add toctree objects."""
 
     @lru_cache(maxsize=None)
-    def generate_header_nav_html(n_links_before_dropdown: int = 5) -> str:
+    def get_id_generator(base_id: str) -> Iterator[str]:
+        for n in count(start=1):
+            if n == 1:
+                yield base_id
+            else:
+                yield f"{base_id}-{n}"
+
+    def create_next_id(base_id: str):
```

Comment:
slightly better name IMO would be something like `uniquify_html_id` or `dedup_html_id` or maybe `get_unique_html_id`

Reply:
How about `unique_html_id`? With "next" I wanted to hint at the sequentiality of the ids, but from an API perspective, that's more of an implementation detail. The only thing the client cares about, I think, is that they get a unique id.

Reply:
that's acceptable. I personally try to make function names verbs (not nouns) but I'm not on a crusade.